Realm/RLMValue.h

```diff
@@ -52,11 +54,15 @@
  `RLMObject
  `RLMObjectId`
  `RLMDecimal128`
+ `RLMDictionary`
+ `RLMArray`
+ `NSArray`
+ `NSDictionary`
  */
 @protocol RLMValue
 
 /// Describes the type of property stored.
-@property (readonly) RLMPropertyType rlm_valueType;
+@property (readonly) RLMMixedValueType rlm_valueType;
```

Comment:
This is a breaking change. It needs to be a new property.

Reply:
Since this is public and the wording we use is around "realm value" and not "mixed", that type should probably be `RLMValueType` or something similar - I don't think we should be throwing in "mixed" in a public API.